### ISSUE TITLE
The inverse of filter

### ISSUE THREAD

**Issue by Integralist** [nvim-neo-tree/neo-tree.nvim#583]
👋🏻 

Just realised it would be great if there was a way to have the inverse of the filter `filter_on_submit` (kinda like the quickfix window has `Cfilter` for "keep anything that matches" vs `Cfilter!` for "excluded anything that matches").

So I had a project with _lots_ of `.go` files and I wanted to excluded the `_test.go` files but couldn't do it because `filter_on_submit` _keeps_ what is matched.

If that's possible already, then apologies.

**Comment by cseickel** [comment]
What we do have is the ability to `hide_by_pattern` which would work for this situation on a permanent basis, and then you can toggle hiding files on and off. This toggle includes all rules for hiding items at the same time, so I am not sure if that is what you are looking for or not. See this section of the help for details:

```
:h neo-tree-filtered-items
```

If you are looking for more of an ad-hoc inverse filter that you can do quickly, then that would be a new feature.

**Comment by Integralist** [comment]
Yeah I think this would be a new feature as I wouldn't want to hide them permanently (even though they could be unhidden fairly quickly). This is likely something I'll find with other file patterns. So a new request is more what I'm suggesting. 

Thanks.
